CHANGELOG.md

```diff
@@ -6,6 +6,7 @@ Mapbox welcomes participation and contributions from everyone.
 #### Features
 
 #### Bug fixes and improvements
+- :warning: Added checks to `DirectionsRoute` and `NavigationRoute` mappers which restrict to map `NavigationRoute` to `DirectionsRoute` and vice versa for some new Directions API features (for now it's EV routing), because original `DirectionsRoute` response requires these kinds of features but one is missing due to mapping process. [#6004](https://github.com/mapbox/mapbox-navigation-android/pull/6004)
```

Comment:
```suggestion
- :warning: Added checks to `DirectionsRoute#toNavigationRoute` and `NavigationRoute#toDirectionsRoute` mappers which restrict mapping `NavigationRoute` to `DirectionsRoute` and vice versa for some Directions API features and properties (currently including only preview EV routing features), because the `DirectionsRoute` cannot carry information necessary to support turn-by-turn navigation when these features are enabled. If you are using EV routing preview feature, make sure to only interact with `MapboxNavigation#requestRoutes(RouteOptions, NavigationRouterCallback)`, `MapboxNavigation#setNavigationRoutes(List<NavigationRoute>)`, and equivalent `NavigationRoute` APIs. [#6004](https://github.com/mapbox/mapbox-navigation-android/pull/6004)
```

Not a blocker for the PR (to save a CI cycle) but @tomaszrybakiewicz @MrSwimmer please adjust the final changelog entry when cutting the release.

Reply:
addressed